### PR TITLE
Issue 19

### PR DESCRIPTION
**Pull request by chottAA5208**

### CODE REVIEW

File: lsnms/nms.py [hunk 1]
```diff
@@ -71,7 +71,8 @@ def _nms(
         keep.append(current_idx)
         to_consider[current_idx] = False
 
-    return np.array(keep)
+    keep = np.array(keep)
+    return np.argwhere(score_mask)[keep]
```
Comment (remydubois) [inline lsnms/nms.py]:
```suggestion
    return np.argwhere(score_mask)[:, 0][keep]
```
Did not know about `argwhere` returning 2d array !

**Review by remydubois — CHANGES_REQUESTED** [review]

**Comment by remydubois** [comment]
The only that is missing for the CI to pass, is that all the tests in `test/test_nms.py` must be modified accordingly. Example with `test_rtree_nms`:

```python
def test_rtree_nms(instances, score_threshold):

    boxes, scores = instances

    # Manually filter instances based on scores because torch's NMS does not do it
    scores_mask = scores > score_threshold
    torch_boxes = boxes[scores_mask]
    torch_scores = scores[scores_mask]

    # Compare against torch
    _k1 = box_ops.nms(torch.tensor(torch_boxes), torch.tensor(torch_scores), 0.5).numpy()
    # argwhere returns array of shape (n, 1)
    k1 = np.argwhere(scores_mask)[_k1, 0]

    # Compare sparse NMS
    k2 = nms(boxes, scores, 0.5, score_threshold)

    assert np.allclose(k1, k2)
```

Et cetera, for the all the other tests of the section.

**Comment by chottAA5208** [comment]
@remydubois - with the latest changes, I believe we are good to go! Let me know if there is anything else I can do to help update this!

**Comment by chottAA5208** [comment]
Remy -


All tests now past! looking forward to seeing these changes added into the next release!
________________________________

**Review by remydubois — APPROVED** [review]

**Comment by chottAA5208** [comment]
Awesome! Looks like i can't merge unless with my current privileges. We will be watching for it to go in! Thanks for working with us (again!)

________________________________

**Comment by chottAA5208** [comment]
WHOO HOOO! Thanks
________________________________

**Comment by remydubois** [comment]
Thanks for the PR !

0.4.1 should be published on  pypi 🎉

**Pull request merged**